### PR TITLE
Add Jenkins for building and deployment 

### PR DESCRIPTION
**Pull request by gfournieriExec**

### CODE REVIEW

File: JenkinsfileDeploy [hunk 1]
```diff
@@ -0,0 +1,97 @@
+node('docker') {
+
+    // Global variables
+    String DOCKER_IMAGE_VERSION = '8491d157'
```
Comment (jeremyjams) [inline JenkinsfileDeploy:4]: Wdyt about adding DOCKER_IMAGE_VERSION to input?

Comment (gfournieriExec) [reply JenkinsfileDeploy:4]: if it's possible to let it idle ? In the way that no input => latest by default or other version

Comment (jeremyjams) [reply JenkinsfileDeploy:4]: imo, if not input, abort
(?)

Comment (gfournieriExec) [reply JenkinsfileDeploy:4]: okay !

File: JenkinsfileBuild [hunk 2]
```diff
@@ -0,0 +1,12 @@
+@Library('global-jenkins-library@2.8.0') _
+
+node('docker') {
+    buildInfo = getBuildInfo()
+    buildSimpleDocker_v2(
+        buildInfo: buildInfo,
+        dockerfileDir: './docker',
+        buildContext: '.',
+        imageprivacy: 'private',
+        dockerImageRepositoryName : 'voucher-subgraph'
```
Comment (jeremyjams) [inline JenkinsfileBuild:10]:
```suggestion
        dockerImageRepositoryName : 'voucher-subgraph-deployer'
```

File: test-stack/docker-compose.yml [hunk 3]
```diff
@@ -88,7 +88,6 @@ services:
       context: ..
       dockerfile: docker/Dockerfile
     environment:
-      RPC_URL: http://bellecour-fork:8545
```
Comment (jeremyjams) [inline test-stack/docker-compose.yml:91]: Why removing `RPC_URL`?

File: JenkinsfileDeploy [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                    -e IPFS_URL='http://$userInput.targetRemoteHost:5001' \
+                    -e GRAPHNODE_URL='http://$userInput.targetRemoteHost:8020' \
+                    -e VERSION_LABEL='$userInput.versionLabel' \
+                    docker-regis.iex.ec/voucher-subgraph-deployer:$DOCKER_IMAGE_VERSION
+            """
+            
+            // Logout to remove docker credentials
+            sh "docker logout docker-regis.iex.ec"
+        }
+        println "The end."
```
Comment (jeremyjams) [inline JenkinsfileDeploy:63]:
```suggestion
```

File: JenkinsfileDeploy [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            IPFS URL: http://$userInput.targetRemoteHost:5001
+            Network Name: $userInput.networkName
+            Voucher Hub Address: $userInput.voucherHubAddress
+            Start Block: $userInput.voucherHubStartBlock
+            Version Label: $userInput.versionLabel
+            Deployer image version: $userInput.dockerImageVersion
+        """
+
+        if (!(userInput.dockerImageVersion?.trim()) 
+                ) {
```
Comment (jeremyjams) [inline JenkinsfileDeploy:33]: Minor, formatting is strange here :)

File: README.md [hunk 6]
```diff
@@ -85,3 +85,20 @@ docker run --rm \
...
+
+## JenkinsfileDeploy
+
+This pipeline is responsible for deploying the voucher-subgraph to the desired environment.
+
+Steps:
+
+1.	User Input: The pipeline prompts the user for input, such as the target host, network name, voucher hub address, and start block.
+3.	Compute Start Block: If a start block isn’t provided, the current block number is fetched from the Bellecour fork (https://bellecour.iex.ec).
+5.	Run Docker Image: The Docker container is executed, deploying the subgraph to the specified Graph Node and IPFS node.
```
Comment (jeremyjams) [inline README.md:104]:
```suggestion
1.	User Input: The pipeline prompts the user for input, such as the target host, network name, voucher hub address, and start block.
2.	Run Docker Image: The Docker container is executed, deploying the subgraph to the specified Graph Node and IPFS node.
```

**Review by jeremyjams — APPROVED** [review]
Good thanks :)

**Pull request merged**